code/game/objects/structures/displaycase.dm

```diff
@@ -114,7 +126,31 @@
 	src.icon = I
 	return
 
-/obj/structure/displaycase/attackby(obj/item/weapon/W, mob/user, params)
+/obj/structure/displaycase/attackby(obj/item/W, mob/user, params)
+	if(is_trophy)
+		if(is_type_in_typecache(W, blacklisted_cargo_types))
+			user << "You think putting [W] in would be a bad idea."
+			return
+		if(!added_roundstart)
+			user << "You've already put something new in this case."
+		var/chosen_plaque = stripped_input(user, "What would you like the plaque to say?", "Trophy Plaque")
+		if(!chosen_plaque)
+			user << "You decide against putting [W] in."
+			return
+		if(showpiece)
+			user << "You press a button, and [showpiece] descends into the floor of the case."
+			SSpersistence.old_trophy_list += "[showpiece.type]|[trophy_message]#"
+			qdel(showpiece)
+			showpiece = null
+		if(user.drop_item())
+			added_roundstart = FALSE
+			user << "You insert [W] into the case."
+			W.forceMove(src)
+			showpiece = W
+			trophy_message = chosen_plaque
+			update_icon()

```

Comment:
This seems unsafe. You could do something like attack with your changeling armblade, swap hands to a pen, and then put the armblade into the display case no problem because the drop_item check dropped your pen, not the armblade you were actually trying to put it. Heck it doesn't even care if you've moved. You could walk to arrivals with the name dialogue open and teleport stuff in remotely.